File: cpp/kalkulator2.cpp
/*
 * kalkulator2.cpp
 */


#include <iostream>

using namespace std;

float dodaj(float a, float b)
{
    return a+b;
}
float odejmij(float a, float b)
{
    return a-b;
}
float pomnoz(float a, float b)
{
    return a*b;
}
float podziel(float a, float b)
{
    return a/b;
}

int main(int argc, char **argv)
{
    
	float liczba1 = 0;
    float liczba2 = 0;
    string znak = "0";
    for(;;)
    {
    cout<<"Podaj liczbe 1: ";
    cin>>liczba1;
    cout<<"Podaj liczbe 2: ";
    cin>>liczba2;
    cout<<"Podaj znak(0 jako exit): ";
    cin>>znak;
    
    while(znak=="/" && liczba2 == 0)
    {
        cout<<"nie dzielimy przez 0"<<endl;
        cout<<"Podaj liczbe 2: ";
        cin>>liczba2;
    }
    
    if(znak == "+")
    {
        cout<<dodaj(liczba1, liczba2)<<endl;
    }
    else if(znak == "-")
    {
        cout<<odejmij(liczba1, liczba2)<<endl;
    }
    else if(znak == "*")
    {
        cout<<pomnoz(liczba1, liczba2)<<endl;
    }
    else if(znak == "/")
    {
        cout<<podziel(liczba1, liczba2)<<endl;
    }
    else if(znak == "0")
    {
        break;
    }
    }
	return 0;
}

File: python/kalkulator2.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-


def suma(a, b):
    return a + b


def roznica(a, b):
    return a - b


def iloczyn(a, b):
    return a * b


def iloraz(a, b):
    return a / b


def main(args):
    a = int(input("Podaj 1. liczbe: "))
    print(a)
    b = int(input("Podaj 2. liczbe: "))

    while b == 0:
        print("nie dzielimy przez 0")
        b = int(input("Podaj 2. liczbe: "))
        print(b)

    print("Suma: ", suma(a,b))
    print("Roznica: ", roznica(a,b))
    print("Iloczyn: ", iloczyn(a,b))
    print("Iloraz: ", iloraz(a,b))

    return 0


if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))
